### PR TITLE
[Android] Implement API XWalkUIClient.onConsoleMessage and teste case

### PR DESCRIPTION
**Pull request by axinging**
Implement API XWalkUIClient.onConsoleMessage for client to process console log.
Add test case for XWalkUIClient.onConsoleMessage.

BUG=XWALK-3305

### CODE REVIEW

File: runtime/android/core_internal/src/org/xwalk/core/internal/XWalkContentsClientBridge.java [hunk 1]
```diff
@@ -260,7 +260,29 @@ public void onLoadResource(String url) {
```
Comment (wang16) [inline runtime/android/core_internal/src/org/xwalk/core/internal/XWalkContentsClientBridge.java]: please change back the file mod.

File: runtime/android/core_internal/src/org/xwalk/core/internal/XWalkUIClientInternal.java [hunk 2]
```diff
@@ -303,6 +303,35 @@ public void onUnhandledKeyEvent(XWalkViewInternal view, KeyEvent event) {
     }
```
Comment (wang16) [inline runtime/android/core_internal/src/org/xwalk/core/internal/XWalkUIClientInternal.java]: ditto

File: runtime/android/core_internal/src/org/xwalk/core/internal/XWalkContentsClientBridge.java [hunk 3]
```diff
@@ -260,7 +260,29 @@ public void onLoadResource(String url) {
...
+                consoleMessageType = XWalkUIClientInternal.ConsoleMessageType.WARNING;
+                break;
+            case ERROR:
+                consoleMessageType = XWalkUIClientInternal.ConsoleMessageType.ERROR;
+                break;
+            default:
+                Log.w(TAG, "Unknown message level, defaulting to DEBUG");
+                break;
+        }
+        return mXWalkUIClient.onConsoleMessage(mXWalkView, consoleMessage.message(),
```
Comment (wang16) [inline runtime/android/core_internal/src/org/xwalk/core/internal/XWalkContentsClientBridge.java]: perform a null pointer checker first.

File: runtime/browser/android/xwalk_web_contents_delegate.cc [hunk 4]
```diff
@@ -162,6 +162,51 @@ void XWalkWebContentsDelegate::RendererResponsive(WebContents* source) {
...
+        source, level, message, line_no, source_id);
+  ScopedJavaLocalRef<jstring> jmessage(ConvertUTF16ToJavaString(env, message));
+  ScopedJavaLocalRef<jstring> jsource_id(
+      ConvertUTF16ToJavaString(env, source_id));
+  int jlevel =
+      web_contents_delegate_android::WEB_CONTENTS_DELEGATE_LOG_LEVEL_DEBUG;
+  switch (level) {
+    case logging::LOG_VERBOSE:
+      jlevel =
+        web_contents_delegate_android::WEB_CONTENTS_DELEGATE_LOG_LEVEL_DEBUG;
```
Comment (wang16) [inline runtime/browser/android/xwalk_web_contents_delegate.cc]: 4 space indent.

File: test/android/core/javatests/src/org/xwalk/core/xwview/test/OnConsoleMessageTest.java [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+    public void testOnConsoleMessageDebug() throws Throwable {
+        String fileContent = getFileContent("console_message.html");
+        int count = mOnConsoleMessageHelper.getCallCount();
+
+        loadDataSync(null, fileContent, "text/html", false);
+        executeJavaScriptAndWaitForResult("doDebug();");
+        mOnConsoleMessageHelper.waitForCallback(count);
+        assertEquals(1, mOnConsoleMessageHelper.getCallCount());
+        assertEquals("debug", mOnConsoleMessageHelper.getMessage());
+    }
```
Comment (wang16) [inline test/android/core/javatests/src/org/xwalk/core/xwview/test/OnConsoleMessageTest.java]: how about add check for the message string?

**Comment by axinging** [comment]
@lincsoon @halton @wang16

File: test/android/core/javatests/src/org/xwalk/core/xwview/test/XWalkViewTestBase.java [hunk 6]
```diff
@@ -89,6 +89,13 @@ public boolean onJavascriptModalDialog(XWalkView view,
         }
 
         @Override
+        public boolean onConsoleMessage(XWalkView view, String message,
+                int lineNumber, String sourceId, ConsoleMessageType messageType) {
+            return mInnerContentsClient.onConsoleMessage(message,lineNumber,sourceId,
+                messageType);
```
Comment (wang16) [inline test/android/core/javatests/src/org/xwalk/core/xwview/test/XWalkViewTestBase.java]: 8 space indent.

**Comment by wang16** [comment]
LGTM after fixing the nit.

**Comment by axinging** [comment]
@wang16  fix the indent of return.

**Comment by lincsoon** [comment]
LGTM

**Pull request merged**